lib/lrud.js

```diff
@@ -214,28 +214,29 @@ const getBlockedExitDirs = (container, candidateContainer) => {
 };
 
 /**
- * Return early if candidate is in the wrong direction
+ * Check if the candidate is in the `exitDir` direction from the rect we're leaving,
+ * with an overlap allowance of entryWeighting as a percentage of the candidate's width.
  *
  * @param {Object} entryRect An object representing the rectangle of the item we're moving to
  * @param {String} exitDir The direction we're moving in
- * @param {Object} exitPoint The first point of the item we're leaving
+ * @param {Object} exitPoint The midpoint of the edge we're leaving
+ * @param {Float} entryWeighting Percentage of the candidate that is allowed to be behind the target
  * @return {Booelan} true if candidate is in the correct dir, false if not
  */
-const isValidCandidate = (entryRect, exitDir, exitPoint) => {
+const isValidCandidate = (entryRect, exitDir, exitPoint, entryWeighting) => {
   if (entryRect.width === 0 && entryRect.height === 0) return false;
+  if (!entryWeighting && entryWeighting != 0) entryWeighting = 0.3;
```

Comment:
One thing otherwise LGTM, is the second part of this check ever gonna be anything but not 0 if `entryWeighting` isn't defined?

Reply:
Yeah, it's to catch NaN which can come from the `parseFloat` that reads the attribute. Both 0 and NaN are falsey so would fall into this, but if they've passed 0 we want to actually use that value and only NaN we want to replace with the default.

Ideally would've just defaulted the argument to 0.3 but NaN slips through that.